src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
Thanks for the reply. So if the player is colliding with a post and the action returns false, addLastActionStatusCollision will return ACTION_STATUS_UNKNOWN.

I must admit that I still don't have a clear understanding of the difference between ACTION_STATUS_MAYBE, ACTION_STATUS_UNKNOWN, and ACTION_STATUS_BAD.

Presumably ACTION_STATUS_BAD means the action failed, but what about the other two? Is there no way to know if an action actually succeded - e.g. the best we can hope for is Maybe?